### PR TITLE
Solana integration with Phantom

### PR DESCRIPTION
**Pull request by leirbag95**
It is now possible to connect to solana through the Phantom wallet.

For the moment only the connection and retrieval of account data is available, the message signature is not yet integrated.
The interaction with solana is done directly from a public rpc https://free.rpcpool.com

The address of the SPL used to calculate the balance of an account is the following:

3UCMiSnkcnkPE1pgQ5ggPCBv6dXgVUy16TmMUe1WpG9x

To know more about solana/web3js 

https://solana-labs.github.io/solana-web3.js/

To know more about connection and interaction with Phantom:

https://docs.phantom.app/

Capture d’écran 2022-06-17 à 15 56 36

### CODE REVIEW

**Comment by hoh** [comment]
@BjrInt can you review this PR ?

File: src/layouts/MainLayout.vue [hunk 1]
```diff
@@ -250,38 +303,101 @@ export default {
...
+          this.$q.notify({
+            type: 'negative',
+            message: 'Non-Ethereum browser detected. You should consider trying MetaMask!'
+          })
+          console.warn('Non-Ethereum browser detected. You should consider trying MetaMask!')
+          return
+        }
+      } else if (wallet == 'phantom') {
+        const isPhantomInstalled = window.solana && window.solana.isPhantom
+        if (isPhantomInstalled) {
```
Comment (BjrInt) [inline src/layouts/MainLayout.vue:355]: No feedback to the user if there is no Phantom extension installed (`$q.notify` an error to the user could be an option). Furthermore the function execution is not stopped if it's the case; provider is still `undefined` leading to an error from line 374

File: src/layouts/MainLayout.vue [hunk 2]
```diff
@@ -41,7 +90,8 @@
             <q-item-label header v-if="item.title" :key="item.title">{{item.title}}</q-item-label>
             <q-item v-for="link of item.items"
                     :key="link.text"
-                    clickable v-ripple
+                    clickable
+                    :disable="isPhantom && link.disable" v-ripple
```
Comment (BjrInt) [inline src/layouts/MainLayout.vue:94]: There is no visual feedback to the user regarding why those buttons are disabled, could be misleading.

**Review by BjrInt — CHANGES_REQUESTED** [review]
Had to manually `yarn lint --fix` to run the code, make sure to use it before you commit (Github action fails on it as well)

**Comment by BjrInt** [comment]
:warning: This branch is forked from #34 and not main!

**Pull request closed without merging**